MekHQ/src/mekhq/gui/dialog/MassRepairSalvageDialog.java

```diff
@@ -634,628 +612,577 @@ public void actionPerformed(ActionEvent e) {
-					mroc.maxSkillCBox.setEnabled(false);
-					mroc.minBTHSpn.setEnabled(false);
-					mroc.maxBTHSpn.setEnabled(false);
-				}
-			}
-		});
-
-		return mroc;
-	}
-
-	private JSpinner createMassRepairSkillBTHSpinner(int selectedValue, boolean enabled, JPanel pnlItems, int rowIdx,
-			int columnIdx) {
-		JSpinner skillBTHSpn = new JSpinner(new SpinnerNumberModel(selectedValue, 1, 12, 1));
-		((JSpinner.DefaultEditor) skillBTHSpn.getEditor()).getTextField().setEditable(false);
-		skillBTHSpn.setEnabled(enabled);
-
-		GridBagConstraints gridBagConstraints = new GridBagConstraints();
-		gridBagConstraints.gridx = columnIdx;
-		gridBagConstraints.gridy = rowIdx;
-		gridBagConstraints.insets = new Insets(0, 5, 0, 5);
-		gridBagConstraints.fill = java.awt.GridBagConstraints.NONE;
-		gridBagConstraints.anchor = java.awt.GridBagConstraints.WEST;
-
-		pnlItems.add(skillBTHSpn, gridBagConstraints);
-
-		return skillBTHSpn;
-	}
-
-	private JComboBox<String> createMassRepairSkillCBox(int selectedValue, boolean enabled, JPanel pnlItems, int rowIdx,
-			int columnIdx) {
-		DefaultComboBoxModel<String> skillModel = new DefaultComboBoxModel<String>();
-		skillModel.addElement(SkillType.getExperienceLevelName(SkillType.EXP_ULTRA_GREEN));
-		skillModel.addElement(SkillType.getExperienceLevelName(SkillType.EXP_GREEN));
-		skillModel.addElement(SkillType.getExperienceLevelName(SkillType.EXP_REGULAR));
-		skillModel.addElement(SkillType.getExperienceLevelName(SkillType.EXP_VETERAN));
-		skillModel.addElement(SkillType.getExperienceLevelName(SkillType.EXP_ELITE));
-		skillModel.setSelectedItem(SkillType.getExperienceLevelName(selectedValue));
-		JComboBox<String> skillCBox = new JComboBox<String>(skillModel);
-		skillCBox.setEnabled(enabled);
-
-		GridBagConstraints gridBagConstraints = new GridBagConstraints();
-		gridBagConstraints.gridx = columnIdx;
-		gridBagConstraints.gridy = rowIdx;
-		gridBagConstraints.insets = new Insets(0, 5, 0, 5);
-		gridBagConstraints.fill = java.awt.GridBagConstraints.NONE;
-
-		pnlItems.add(skillCBox, gridBagConstraints);
-
-		return skillCBox;
-	}
-
-	private JCheckBox createMassRepairOptionItemBox(String text, String toolTipText, String name, boolean selected,
-			JPanel pnlItems, int rowIdx, int columnIdx) {
-		JCheckBox optionItemBox = new JCheckBox();
-		optionItemBox.setText(text);
-		optionItemBox.setToolTipText(toolTipText);
-		optionItemBox.setName(name);
-		optionItemBox.setSelected(selected);
-		if (name.equals("massRepairItemPod") && !isModeWarehouse()) {
+        return pnlOptions;
+    }
+
+    private MassRepairOptionControl createMassRepairOptionControls(int type, String text,String tooltipText,
+                                                                   String activeBoxName, JPanel pnlItems,
+                                                                   int rowIdx) {
+        MassRepairOption mro = null;
+
+        for (MassRepairOption massRepairOption : campaignOptions.getMassRepairOptions()) {
```

Comment:
Isn't this going to throw an exception of getMassRepairOptions returns null?

Reply:
Fixed it on one end, and I think I forgot to fix it on the other end... give me five and I'll double check

Reply:
Yep, forgot to fix it on the other end... That's now fixed